index.js

```diff
@@ -1,121 +1,74 @@
-'use strict';
 const request = require('request-promise-native');
 
 const role_to_use = process.env.ROLE;
 const couch_url = process.env.COUCH_URL;
-let compiledUrl = {};
 
-if(role_to_use === undefined){
+if(role_to_use === undefined) {
     console.log('ROLE is not set, please set it and try again');
     process.exit(1);
 }
-if(couch_url === undefined){
+if(couch_url === undefined) {
     console.log('COUCH_URL is not set, please set it and try again');
     process.exit(1);
 }
 
-const handleError = function(e){
-    if (e.statusCode === 401) {
-        console.log('Bad authentication for CouchDB. Check that COUCH_URL has correct usernames and passwords.');
-        process.exit(1);
-    } else {
-        console.log('Error connecting to CouchDB: ' + e.message);
-        process.exit(1);
-    }
-}
-
-const getAllUsers = function (){
+const compileUrl = (path) => {
     try {
-        compiledUrl = new URL('/_users/_all_docs', couch_url);
+        return new URL(path, couch_url).href;
     } catch(e) {
-        console.log('Error while creating url: ', e.message);
-        process.exit(1);
```

Comment:
Crashing an execution like this is frowned upon so it's better to have a single `exit` call in a known place, rather than using it throughout.

Reply:
see other comment about CLI use - but I see where this is going...